### ISSUE TITLE
Fails to start with PasswordAuthenticator

### ISSUE THREAD

**Issue by MartinEmrich** [exoscale/pithos#87]
Using pithos with PasswordAuthenticator enabled fails for me:

`
invalid or incomplete configuration:  com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/127.0.0.1:9042 (com.datastax.driver.core.ConnectionException: [localhost/127.0.0.1:9042] Unexpected error during transport initialization (java.lang.NullPointerException)))
ERROR [2016-04-26 08:04:25,842] main - io.pithos.config invalid or incomplete configuration
com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/127.0.0.1:9042 (com.datastax.driver.core.ConnectionException: [localhost/127.0.0.1:9042] Unexpected error during transport initialization (java.lang.NullPointerException)))
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:227) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:82) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1307) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at com.datastax.driver.core.Cluster.init(Cluster.java:159) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at com.datastax.driver.core.Cluster.connect(Cluster.java:249) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at qbits.alia$connect.invoke(alia.clj:196) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at io.pithos.store$cassandra_store.invoke(store.clj:32) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at io.pithos.bucket$cassandra_bucket_store.invoke(bucket.clj:90) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at clojure.lang.Var.invoke(Var.java:379) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at io.pithos.config$instantiate.invoke(config.clj:89) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at io.pithos.config$get_instance.invoke(config.clj:97) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at clojure.lang.AFn.applyToHelper(AFn.java:156) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at clojure.lang.AFn.applyTo(AFn.java:144) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at clojure.core$apply.invoke(core.clj:632) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at clojure.core$update_in.doInvoke(core.clj:5923) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at clojure.lang.RestFn.invoke(RestFn.java:467) [pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at io.pithos.config$init.invoke(config.clj:159) ~[pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at io.pithos$_main.doInvoke(pithos.clj:66) [pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at clojure.lang.RestFn.applyTo(RestFn.java:137) [pithos-0.7.5-SNAPSHOT-standalone.jar:na]
        at io.pithos.main(Unknown Source) [pithos-0.7.5-SNAPSHOT-standalone.jar:na]
`
I added username and password to the kestore, bucketstore, metastore aund storage-classes.

Is this supposed to work?

**Comment by veselov** [comment]
I had the same problems. Unless it's been implemented recently, it doesn't work out of the box. I fixed this myself:

```
diff --git a/src/io/pithos/store.clj b/src/io/pithos/store.clj
index cf443e2..9ef3344 100644
--- a/src/io/pithos/store.clj
+++ b/src/io/pithos/store.clj
@@ -17,7 +17,7 @@
 (defn cassandra-store
   "Connect to a cassandra cluster, and use a specific keyspace.
    When the keyspace is not found, try creating it"
-  [{:keys [cluster keyspace hints repfactor]}]
+  [{:keys [cluster keyspace hints repfactor username password]}]
   (debug "building cassandra store for: " cluster keyspace hints)
   (let [hints   (or hints
                     {:replication {:class             "SimpleStrategy"

diff --git a/src/io/pithos/store.clj b/src/io/pithos/store.clj
index 071aad0..cf443e2 100644
--- a/src/io/pithos/store.clj
+++ b/src/io/pithos/store.clj
@@ -23,7 +23,9 @@
                     {:replication {:class             "SimpleStrategy"
                                    :replication_factor (or repfactor 1)}})
         cluster (if (sequential? cluster) cluster [cluster])
-        session (-> (alia/cluster {:contact-points cluster}) (alia/connect))]
+        session (-> (alia/cluster {:credentials { :user username :password password}
+                                   :contact-points cluster})
+                    (alia/connect))]

```
I didn't submit this as a PR as I have no knowledge of clojure, and with this diff, omitting username/password in config leads to NPE.

**Comment by pyr** [comment]
Hi!

This is now implemented in pithos 0.7.5, sorry to have kept you waiting!
